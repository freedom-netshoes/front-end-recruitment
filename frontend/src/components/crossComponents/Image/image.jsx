import React, { Component } from 'react'
import CSSModules from 'react-css-modules'
import { PropTypes } from 'prop-types'
import styles from './image.less'
import { appSettings } from '../../../resources/appSettings'

@CSSModules(styles)
export default class Image extends Component {
  static defaultProps = {
    sku: '',
    title: '',
    isThumb: false
  };

  static propTypes = {
    sku: PropTypes.string,
    title: PropTypes.string,
    isThumb: PropTypes.bool
  };

  render () {
    const {
      title,
      sku
    } = this.props
    const partialUrl = `${appSettings.imgUrl}${sku}`
    if (!this.props.isThumb) {
      return (
        <figure {...{ className: 'photo' }} >
          <img {... { src: `${partialUrl}.jpg`, alt: title }} />
        </figure>
      )
    }
    return (
      <figure {...{ className: 'photo' }} >
        <img {... { src: `${partialUrl}_thumb.jpg`, alt: title }} />
      </figure>
    )
  }
}
