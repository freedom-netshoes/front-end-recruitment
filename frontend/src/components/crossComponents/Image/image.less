:global {
    #products {
        .photo {
            /*height: 230px;
            width: 180px;*/
        }
    }
}
