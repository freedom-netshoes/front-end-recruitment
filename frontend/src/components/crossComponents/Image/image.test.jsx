/* global describe, it, expect */
import React from 'react'
import Image from './image'
import { shallow } from 'enzyme'

describe('Testing <Image /> component', () => {
  it('renders the Image component', () => {
    const component = shallow(< Image />)
    expect(component).toHaveLength(1)
  })

  it('Testing <Image /> snapshop', () => {
    const component = shallow(<Image {...{
      sku: '18644119330491312',
      title: 'Camisa Nike Corinthians II',
      isThumb: false
    }} />)
    expect(component).toMatchSnapshot()
  })
})
