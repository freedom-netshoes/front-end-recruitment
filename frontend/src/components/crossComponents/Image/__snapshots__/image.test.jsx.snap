// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Image /> component Testing <Image /> snapshop 1`] = `
<figure
  className="photo"
>
  <img
    alt="Camisa Nike Corinthians II"
    src="images/18644119330491312.jpg"
  />
</figure>
`;
